### PR TITLE
Sandrine's todo app

### PR DESCRIPTION
**Pull request by MindstormingAB**
Link to my deployed project: https://sandrines-todo.netlify.app/

My reflections on how this week's project turned out: I found very challenging to use redux-toolkit to create slices, reducers and actions by myself

Things I'd like to have clarified or explained in more detail: I created priority levels for the todos and it works fine to put a priority level when creating a new todo but I didn't manage to change the priority of an existing todo.

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 <html lang="en">
 
-  <head>
-    <meta charset="utf-8" />
-    <meta name="viewport" content="width=device-width, initial-scale=1" />
-    <!--
+<head>
+  <meta charset="utf-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <!--
```
Comment (pwangy) [inline code/public/index.html:7]: Housekeeping note: You may want to remove these commented out lines.

File: code/src/App.js [hunk 2]
```diff
@@ -1,9 +1,25 @@
 import React from 'react'
+import { Provider } from 'react-redux'
+import { configureStore, combineReducers } from '@reduxjs/toolkit'
+
+import { Header } from 'components/Header'
+import { Input } from 'components/Input'
+import { TaskList } from 'components/TaskList'
+
```
Comment (pwangy) [inline code/src/App.js:8]: not sure if this extra blank line is needed.

File: code/src/components/Input.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components/';
+
+import { tasks } from 'reducers/tasks';
+
```
Comment (pwangy) [inline code/src/components/Input.js:6]: extra line break

File: code/src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  font-size: 20px;
+  font-weight: bold;
+`;
+
+const Tasks = styled.div`
+  // font-size: 16px;
+  text-align: end;
+`;
+
+const Date = styled.div`
```
Comment (pwangy) [inline code/src/components/Header.js:25]: looks like no special styling is used here. could this be replaced with a plain div tag?

**Review by pwangy — APPROVED** [review]
Hi Sandrine, lovely job of creating a clean App this week and some very tidy code to go with it. ✨ 

I left comments only where I saw inconsistencies or opportunities where code could be cleaner. It appears that there's a little inconsistency with the usage of semicolons (none in App.js and Tasks.js, but they appear throughout the rest of the code). So a gentle reminder to pick one style and stick with it throughout the entire app. 

Remembering that you mentioned your time restriction for this week, you've done a splendid job of hitting all the blue goals, implementing a category  by allowing for a priority setting, AND creating the challenge of being able to change the priority after adding the item to the list. Ambitious!

File: code/src/App.js [hunk 5]
```diff
@@ -1,9 +1,25 @@
 import React from 'react'
+import { Provider } from 'react-redux'
+import { configureStore, combineReducers } from '@reduxjs/toolkit'
+
+import { Header } from 'components/Header'
```
Comment (EmOlsz) [inline code/src/App.js:5]: Next time let's remember about `./` in path

File: code/src/components/Input.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import add from 'assets/add.svg';
+
+const InputContainer = styled.div`
+  display: grid;
+  grid-template-columns: 30px 1fr 80px;
+  align-items: center;
+  padding: 20px;
+  color: #dcd7e1;
+  // font-size: 20px;
```
Comment (EmOlsz) [inline code/src/components/Input.js:15]: I see you commented a lot of font-sizes, they should be deleted

File: code/src/components/Input.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  const onDescriptionChange = (event) => {
+    setDescription(event.target.value);
+  };
+
+  const onPriorityChange = (event) => {
+    setPriority(event.target.value);
+  };
+
+  const onItemAdd = () => {
+    if (description !== '') {
```
Comment (EmOlsz) [inline code/src/components/Input.js:57]: Great job :)

File: code/src/components/Input.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+      <PlusSign
+        type='button'
+        onClick={onItemAdd}
+      >
+        <img height="16" src={add} alt="Add task" />
+      </PlusSign>
+      <InputText
+        rows='1'
+        placeholder='Add task'
+        required
```
Comment (EmOlsz) [inline code/src/components/Input.js:75]: No need for this attribute if we handle content by JS

File: code/src/reducers/tasks.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  ]
+}
+
+export const tasks = createSlice({
+  name: 'tasks',
+  initialState,
+  reducers: {
+    addItem: (state, action) => {
+      const newItem = {
+        id: Date.now(),
```
Comment (EmOlsz) [inline code/src/reducers/tasks.js:32]: It's bit inconsistent to initialise 3 items from initialState with ids as numbers, and later on add new items with ids as dates

File: code/src/reducers/tasks.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+      if (clickedItem) {
+        clickedItem.completed = !clickedItem.completed;
+      }
+    },
+
+    // updatePriority: (state, action) => {
+    //   console.log(action.payload);
+    //   const changedItem = state.items.find((item) => item.id === action.payload.id);
+    //   changedItem.priority = action.payload.priority;
+    // },
```
Comment (EmOlsz) [inline code/src/reducers/tasks.js:52]: Commented code should be deleted

**Pull request closed without merging**